Content.Server/Wieldable/WieldableSystem.cs

```diff
@@ -0,0 +1,280 @@
+                AttemptUnwield(uid, component, args.User);
+        }
+
+        public bool CanWield(EntityUid uid, WieldableComponent component, IEntity user)
+        {
+            // Do they have enough hands free?
+            if (!ComponentManager.TryGetComponent<HandsComponent>(user.Uid, out var hands))
+            {
+                user.PopupMessage(Loc.GetString("wieldable-component-no-hands"));
+                return false;
+            }
+
+            if (hands.GetFreeHands() < component.FreeHandsRequired)
+            {
+                // TODO FLUENT need function to change 'hands' to 'hand' when there's only 1 required
+                user.PopupMessage(Loc.GetString("wieldable-component-not-enough-free-hands",
+                    ("number", component.FreeHandsRequired),
+                    ("item", EntityManager.GetEntity(uid))));
+                return false;
+            }
+
+            // Is it.. actually in one of their hands?
+            if (!hands.TryGetHandHoldingEntity(EntityManager.GetEntity(uid), out var _))
+            {
+                user.PopupMessage(Loc.GetString("wieldable-component-not-in-hands",
+                    ("item", EntityManager.GetEntity(uid))));
+                return false;
+            }
+
+            // Seems legit.
+            return true;
+        }
+
+        /// <summary>
+        ///     Attempts to wield an item, creating a DoAfter..
+        /// </summary>
+        public void AttemptWield(EntityUid uid, WieldableComponent component, IEntity user)
+        {
+            if (!CanWield(uid, component, user))
+                return;
+            var ev = new BeforeWieldEvent();
+            RaiseLocalEvent(uid, ev, false);
+            var used = EntityManager.GetEntity(uid);
+
+            if (ev.Cancelled) return;
+
+            var doargs = new DoAfterEventArgs(
+                user,
+                component.WieldTime,
+                default,
+                used
+            )
+            {
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                BreakOnTargetMove = true,
+                TargetFinishedEvent = new ItemWieldedEvent(user),
+                UserFinishedEvent = new WieldedItemEvent(used)
+            };
+
+            _doAfter.DoAfter(doargs);
+        }
+
+        /// <summary>
+        ///     Attempts to unwield an item, with no DoAfter.
+        /// </summary>
+        public void AttemptUnwield(EntityUid uid, WieldableComponent component, IEntity user)
+        {
+            var ev = new BeforeUnwieldEvent();
+            RaiseLocalEvent(uid, ev, false);
+            var used = EntityManager.GetEntity(uid);
+
+            if (ev.Cancelled) return;
+
+            var targEv = new ItemUnwieldedEvent(user);
+            var userEv = new UnwieldedItemEvent(used);
+
+            RaiseLocalEvent(uid, targEv, false);
+            RaiseLocalEvent(user.Uid, userEv, false);
```

Comment:
When would another system subscribe to these events? Currently these could just be direct calls to functions in WieldableSystem, right?

Reply:
1) ecs doafter requires an event
2) i cant think of anything immediate that would subscribe to these but i'm sure they exist